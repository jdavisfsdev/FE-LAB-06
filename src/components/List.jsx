import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Character from './Character';

export default function List({ characters }) {
  return (
    <ul data-testid="ul">
      {characters.map((character) => (
        <li key={character._id}> <Link to={`/${character._id}`}>
          <Character {...character} />
        </Link>
        </li>
      ))}
    </ul>
  );
}

Character.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }))
};
