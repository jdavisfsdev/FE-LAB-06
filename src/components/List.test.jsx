import React from 'react';
import { render } from '@testing-library/react';
import List from './List';

describe('List component', () => {
  it('render list', () => { 
    const characters = [];
    const { asFragment } = render(
      <List characters={characters}/>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
