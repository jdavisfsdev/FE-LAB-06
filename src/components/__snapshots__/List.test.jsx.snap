// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component render list 1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;
